core/domain/classifier_services.py

```diff
@@ -100,8 +101,32 @@ def classify_string_classifier_rule(state, normalized_answer):
             training_examples.extend([
                 [doc, [str(answer_group_index)]]
                 for doc in classifier_rule_spec.inputs['training_data']])
-    if len(training_examples) > 0:
+
+    exploration = exp_services.get_exploration_by_id(exp_id,
+                                                     version=exp_version)
+    algorithm_id = feconf.INTERACTION_CLASSIFIER_MAPPING['TextInput']
+    if state.classifier_model_id:
+        classifier = get_classifier_by_id(state.classifier_model_id)
+        sc_dict = classifier.cached_classifier_data
+        sc = classifier_registry.Registry.get_classifier_by_algorithm_id(
+            feconf.INTERACTION_CLASSIFIER_MAPPING['TextInput'])
+        sc.from_dict(sc_dict)
+
+    else:
+        sc = classifier_registry.Registry.get_classifier_by_algorithm_id(
+            feconf.INTERACTION_CLASSIFIER_MAPPING['TextInput'])
         sc.train(training_examples)
+        cached_classifier_data = sc.to_dict()
+        schema_version = exploration.states_schema_version
+        classifier = classifier_domain.Classifier('0', exploration.id,
+                                                  exp_version,
+                                                  state_name, algorithm_id,
+                                                  cached_classifier_data,
+                                                  schema_version)
+        classifier_id = save_classifier(classifier)
+        state.classifier_model_id = classifier_id
```

Comment:
@anmolshkl I tried something like this: 
```
change_list = [{
            'cmd': 'edit_state_property',
            'state_name': state_name,
            'property_name': 'classifier_model_id',
            'new_value': classifier_id
        }]
        print exp_services.get_exploration_by_id(exp_id).states[state_name].classifier_model_id
        exp_services.update_exploration(
            owner_id, exp_id, change_list, '')
```
The field isn't getting updated though. We'll have to dig a bit deep here.  Do you have any suggestions?